src/deferred.rs

```diff
@@ -0,0 +1,140 @@
+use std::mem;
+use std::ptr;
+
+/// Number of words a piece of `Data` can hold.
+///
+/// Three words should be enough for the majority of cases. For example, you can fit inside it the
+/// function pointer together with a fat pointer representing an object that needs to be destroyed.
+const DATA_WORDS: usize = 3;
+
+/// Some space to keep a `FnOnce()` object on the stack.
+type Data = [usize; DATA_WORDS];
```

Comment:
What about inlining this definition?

Reply:
You mean, why not remove this type? Because then things like
```rust
if size <= mem::size_of::<Data>() && align <= mem::align_of::<Data>() {
```

would become
```rust
if size <= mem::size_of::<[usize; DATA_WORDS]>() && align <= mem::align_of::<[usize; DATA_WORDS]>() {
```